src/components/Table/GridTable.tsx

```diff
@@ -270,141 +259,73 @@ export function GridTable<R extends Kinded, X extends Only<GridTableXss, X> = an
   const expandedColumnIds: string[] = useComputed(() => tableState.expandedColumnIds, [tableState]);
   const columnSizes = useSetupColumnSizes(style, columns, resizeTarget ?? resizeRef, expandedColumnIds);
 
-  // Make a single copy of our current collapsed state, so we'll have a single observer.
-  const collapsedIds = useComputed(() => tableState.collapsedIds, [tableState]);
-  const sortState = useComputed(() => toJS(tableState.sortState), [tableState]);
-
-  const maybeSorted = useMemo(() => {
-    if (sortOn === "client" && sortState) {
-      // If using client-side sort, the sortState use S = number
-      return sortRows(columns, rows, sortState, caseSensitive);
-    }
-    return rows;
-  }, [columns, rows, sortOn, sortState, caseSensitive]);
-
-  const [keptGroupRow, keptDataRows] = useComputed(
-    () => [tableState.keptRowGroup, tableState.keptRows as GridDataRow<R>[]],
-    [tableState],
-  );
-  // Sort the `keptSelectedDataRows` separately because the current sorting logic sorts within groups and these "kept" rows are now displayed in a flat list.
-  // It could also be the case that some of these rows are no longer in the `props.rows` list, and so wouldn't be sorted by the `maybeSorted` logic above.
-  const sortedKeptSelections = useMemo(() => {
-    if (sortOn === "client" && sortState && keptDataRows.length > 0) {
-      return sortRows(columns, keptDataRows, sortState, caseSensitive);
-    }
-    return keptDataRows;
-  }, [columns, sortOn, sortState, caseSensitive, keptDataRows]);
-
   // Flatten, hide-if-filtered, hide-if-collapsed, and component-ize the sorted rows.
-  let [headerRows, visibleDataRows, totalsRows, expandableHeaderRows, keptSelectedRows, filteredRowIds]: [
+  const [tableHeadRows, visibleDataRows, keptSelectedRows, tooManyClientSideRows]: [
     RowTuple<R>[],
     RowTuple<R>[],
     RowTuple<R>[],
-    RowTuple<R>[],
-    RowTuple<R>[],
-    string[],
-  ] = useMemo(() => {
-    const hasExpandableHeader = maybeSorted.some((row) => row.id === EXPANDABLE_HEADER);
-    const makeRowComponent = (
-      row: GridDataRow<R>,
-      level: number,
-      isKeptSelectedRow: boolean = false,
-      isLastKeptSelectionRow: boolean = false,
-    ) => (
-      <Row
-        key={`${row.kind}-${row.id}`}
-        {...{
-          as,
-          columns,
-          row,
-          style,
-          rowStyles,
-          columnSizes,
-          level,
-          getCount,
-          api,
-          cellHighlight: "cellHighlight" in maybeStyle && maybeStyle.cellHighlight === true,
-          omitRowHover: "rowHover" in maybeStyle && maybeStyle.rowHover === false,
-          hasExpandableHeader,
-          isKeptSelectedRow,
-          isLastKeptSelectionRow,
-        }}
-      />
-    );
+    boolean,
+  ] = useComputed(() => {
```

Comment:
Changed this to a `uesComputed` that essentially reacts to `header` `RowState.selfAndMaybeChildren` changing (which itself changes if any top-level rows have their `selfAndMaybeChildren` change, etc)